provider/openstack/provider.go

```diff
@@ -1265,6 +1264,11 @@ func (e *Environ) startInstance(
 		Metadata:           args.InstanceConfig.Tags,
 		AvailabilityZone:   args.AvailabilityZone,
 	}
+
+	if args.Constraints.HasImageID() {
```

Comment:
Shouldn't this be part of findInstanceSpec?

Reply:
So, the json field that openstack (nova) needs to force the selection of a image id is `imageRef`, and this is serialized using the `ImageId` field in `nova.RunServerOpts` (go-goose):
```
type RunServerOpts struct {
	Name                string               `json:"name"`                              // Required
	FlavorId            string               `json:"flavorRef"`                         // Required
	ImageId             string               `json:"imageRef,omitempty"`                // Optional
	...
```
and since this struct is only instantiated in line 1258 (`opts`) I just added this field if needed (i.e. if the user has indeed provided an image-id, otherwise leaving it empty).
Furthermore, we don't use the spec's Image.Id for this `RunServerOpts` here, it's only used when configuring root disk (see your previous comment), so I don't see how I could add it properly on `findInstanceSpec()`.

Reply:
I thought the plan was to move all of the image_id pieces to inside of configureRootDisk?

Reply:
ok yes indeed, I moved it to `configureRootDisk`, I think I didn't understood your advice the first time